insolver/feature_monitoring/homogeneity_report.py

```diff
@@ -0,0 +1,313 @@
+    fig = py.graph_objects.Figure(data=fig1.data + fig2.data)
+
+    # add details
+    fig.update_layout(
+        autosize=False,
+        width=830,
+        height=650,
+        legend=dict(x=0.8, y=0.95, traceorder='normal', font=dict(color='black', size=16)),
+    )
+
+    if offline:
+        return py.offline.plot(fig, include_plotlyjs=False, output_type='div')
+    else:
+        return fig
+
+
+class HomogeneityReport:
+    """
+    This class builds report for two dataframes containing same features.
+    Report consists of homogeneity checks between feature's condition in first and in second frame.
+    We run homogeneity tests for each feature and render report with distributions if necessary.
+
+    The class supports wide configuration of running tests. It takes config. dictionary with which
+    has feature names as keys. Each value for certain feature is sub-dictionary of properties.
+    List of supported properties:
+        feature_type (str): stat. type of features. ('continuous'/'discrete')
+        pval_thresh (float): threshold for pvalue for making conclusions in tests.
+        samp_size (int): size of sub-samples in bootstrap.
+        bootstrap_num (int): number of bootstrap restarts.
+        psi_bins (int): number of bins for psi calculation (only for 'continuous').
+        chart_bins (int): number of bins to draw hists (only for 'continuous').
+        chart_limits (iterable of two floats/ints): custom limits for continuous feature charts
+        ('only for continuous').
+
+    Only 'feature_type' ('continuous'/'discrete') must be specified for each feature,
+    other properties can be missing.
+
+    Default parameters for each property:
+    pval_thresh=0.05, samp_size=500, bootstrap_num=100,
+    psi_bins=20, chart_bins=15, chart_limits=[min(x1, x2), max(x1, x2)].
+
+    Parameters:
+        config_dict (dict): dict. with feature properties (see description).
+    """
+
+    def __init__(self, config_dict: dict = {}):
+        """
+        Raises:
+            KeyError: If it is not specified whether certain feature is continuous or discrete.
+            ValueError: if 'feature_type' is not 'continuous' or 'discrete'.
+        """
+
+        for feat in config_dict:
+            properties = config_dict[feat]
+            if 'feature_type' not in properties:
+                raise KeyError(f"Type of {feat} feature is not found in 'config_dict'.")
+            elif properties['feature_type'] not in ['continuous', 'discrete']:
+                raise ValueError(f"Types of features must be 'continuous' or 'discrete'. Invalid type for {feat}.")
+        self.config_dict = config_dict
+        self.features = config_dict.keys()
+
+    def build_report(
+        self,
+        df1,
+        df2,
+        dropna: bool = False,
+        name1: str = 'Base subset',
+        name2: str = 'Current subset',
+        render: bool = False,
+        report_path: str = None,
```

Comment:
`None` is not a `str`. Consider specifying another default parameter or type annotation